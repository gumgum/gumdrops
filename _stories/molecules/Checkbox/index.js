import React from 'react';
import { text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import FormGroup from '../../../components/molecules/FormGroup';
import Checkbox from '../../../components/molecules/Checkbox';

const sizeOptions = {
    sm: 'sm',
    xs: 'xs',
    'No Value': ''
};

const component = () => (
    <FormGroup>
        <Checkbox
            label={text('Label', 'party parrots')}
            size={optionalSelect('Size', sizeOptions, '')}
            className={text('ClassName', '')}
            style={object('Style', {})}
            onChange={action('checkbox_checked')}
        />
    </FormGroup>
);

export default [readme, component];
