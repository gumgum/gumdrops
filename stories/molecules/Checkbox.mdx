import { Canvas, Source } from '@storybook/blocks';
import { ArgsTable } from '@storybook/addon-docs';

import * as CheckboxStories from './Checkbox.stories';
import Checkbox from '../../components/molecules/Checkbox';

# Checkbox

The `<Checkbox>` component can be wrapped with the `<FormGroup>` component. It can be used as a controlled or uncontrolled component.

## Controlled vs Uncontrolled

This component can be controlled or uncontrolled. This component accepts any prop that you may want to add. The only drawback is that there is not validation for unlisted props. **It is highly recommended that you use controlled components in React.**

_Recommended Props_: `name`, `checked`, `defaultChecked`, `onChange`

### Controlled example

**Example with reusable checkbox handler**:

```jsx
const [pineapple, setPineapple] = useState(false);
  return (
    <FormGroup>
      <Checkbox
            name="pineapple"
            label="pineapple on pizza?"
            checked={ pineapple }
            onChange={e => {
              setPineapple(e.target.checked) }
            }
        />
      </FormGroup>
  );
};
```

### Uncontrolled example

```jsx
_handleSubmit = (e) => {
    e.preventDefault();
    const {
        pineapple,
        otherName,
        otherName2
    } = e.target.elements;
    console.log(pineapple.checked);
};

render() {
    return (
        <form onSubmit={ this._handleSubmit }>
            <FormGroup>
                { /* Pass defaultChecked if you want
                the input to be checked initially */ }
                <Checkbox
                    name="pineapple"
                    label="pineapple on pizza?"
                    defaultChecked
                />
            </FormGroup>
        </form>
    );
}
```

## Component Example

<Canvas of={CheckboxStories.Default} />

## Component Example Source

<Source of={CheckboxStories.Default} />

## Checkbox Props

<ArgsTable of={Checkbox} />
