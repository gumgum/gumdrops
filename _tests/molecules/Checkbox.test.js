import renderer from 'react-test-renderer';
import React from 'react';
import Checkbox from '../../components/molecules/Checkbox';

const defaultProps = {
    label: 'Check Me',
    size: 'xs',
    className: 'custom-class',
    style: { width: 140 }
};

test('Expect <CardImage> to render properly', () => {
    const tree = renderer.create(<Checkbox {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <CardImage> to render with other props', () => {
    const props = {
        ...defaultProps,
        onChange: jest.fn(),
        value: 'foo'
    };
    const tree = renderer.create(<Checkbox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
