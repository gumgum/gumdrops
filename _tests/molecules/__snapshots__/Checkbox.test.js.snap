// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <CardImage> to render properly 1`] = `
<div
  className="gds-form-group__checkbox custom-class gds-form-group__checkbox--xs"
  style={
    {
      "width": 140,
    }
  }
>
  <label
    className="gds-form-group__checkbox-label"
  >
    <input
      className="gds-form-group__checkbox-input"
      type="checkbox"
    />
    <span
      className="gds-form-group__checkbox-indicator"
    />
    Check Me
  </label>
</div>
`;

exports[`Expect <CardImage> to render with other props 1`] = `
<div
  className="gds-form-group__checkbox custom-class gds-form-group__checkbox--xs"
  style={
    {
      "width": 140,
    }
  }
>
  <label
    className="gds-form-group__checkbox-label"
  >
    <input
      className="gds-form-group__checkbox-input"
      onChange={[MockFunction]}
      type="checkbox"
      value="foo"
    />
    <span
      className="gds-form-group__checkbox-indicator"
    />
    Check Me
  </label>
</div>
`;
